_posts/2016-10-06-BDD-tests-for-Hadoop-with-Cucumber-part-I.md

```diff
@@ -0,0 +1,310 @@
+    maven{ url 'http://conjars.org/repo/' }
+}
+
+configurations {
+    providedCompile
+
+    cucumberRuntime {
+        exclude group: 'commons-httpclient'
+        extendsFrom testRuntime
+    }
+}
+
+sourceSets {
+    main {
+        compileClasspath += configurations.providedCompile
+    }
+
+    test {
+        output.resourcesDir = 'build/classes/test/resources'
+        output.classesDir   = 'build/classes/test/java'
+    }
+}
+
+dependencies {
+    // install cascading
+    compile( group: 'cascading', name: 'cascading-core', version: cascadingVersion )
+    compile( group: 'cascading', name: 'cascading-local', version: cascadingVersion )
+    compile( group: 'cascading', name: 'cascading-hadoop2-mr1', version: cascadingVersion )
+    compile( group: 'cascading', name: 'cascading-hadoop2-tez', version: cascadingVersion )
+
+    compile( group: 'com.hotels', name: 'corc-cascading', version: '1.0.0' )
+
+    // install hadoop
+    providedCompile( group: 'org.apache.hadoop', name: 'hadoop-common', version: hadoop2Version )
+    providedCompile( group: 'org.apache.hadoop', name: 'hadoop-client', version: hadoop2Version )
+    providedCompile( group: 'org.apache.hadoop', name: 'hadoop-mapreduce-client-core', version: hadoop2Version )
+
+    compile( group: 'org.apache.hive', name: 'hive-exec', version: hiveVersion )
+    compile( group: 'org.apache.hive', name: 'hive-serde', version: hiveVersion )
+
+    cucumberRuntime files("${jar.archivePath}")
+    testCompile( group: 'org.apache.hadoop', name: 'hadoop-common', version: hadoop2Version )
+    testCompile( group: 'org.apache.hadoop', name: 'hadoop-client', version: hadoop2Version )
+    testCompile( group: 'org.apache.hadoop', name: 'hadoop-mapreduce-client-core', version: hadoop2Version )
+
+    // install cucumber
+    testCompile( group: 'info.cukes', name: 'cucumber-java', version: '1.2.4' )
+    testCompile( group: 'info.cukes', name: 'cucumber-junit', version: '1.2.4' )
+    testCompile( group: 'info.cukes', name: 'cucumber-picocontainer', version: '1.2.4' )
+
+    // install chunk templates for test data generation
+    testCompile( group: 'com.x5dev', name: 'chunk-templates', version: '3.1.0' )
+}
+
+
+task cucumber() {
+    dependsOn assemble, compileTestJava, processTestResources
+    doLast {
+        javaexec {
+            main = "cucumber.api.cli.Main"
+            classpath = configurations.cucumberRuntime + sourceSets.test.runtimeClasspath
+            args = ["--plugin", "pretty",  "--tags", "~@ignore", "--glue", "com.steps", "src/test/resources"]
+        }
+    }
+}
+
+
+jar {
+    description = "Assembles a Hadoop ready jar file"
+    doFirst {
+        into( 'lib' ) {
+            from configurations.compile
+        }
+    }
+
+    manifest {
+
+    }
+}
+
+{% endraw %}
+{% endhighlight %}
+
+As you can see there are two custom tasks - `cucumber` and `jar`.  First one run cucumber tests and the second one compile the source to the production jar-file.
```

Comment:
First one run cucumber tests and the second one compile the source to the production jar-file.
First one run**s** cucumber tests and the second one compile**s** the source to the production jar-file.